### PR TITLE
Add positive grading to assessments

### PR DESCRIPTION
**Pull request by victorhuangwq**
## Description

Adds ability to do positive grading to assignments as well as the following sub-features:
- Warning when switching between positive and negative if there's currently an annotation
- Add an indication about whether it's positive or negative grading in the annotation box
- Add `is_positive_grading` to export assessments

## Motivation and Context

Iliano requested the feature to be added as changing 122 rubrics from positive to negative has been difficult for their class.

## How Has This Been Tested?

**Grading style testing**

Tested negative grading:
image
image

Tested positive grading:
image
image

**Tested Annotations Warning**
- Tested changing when there are annotations > warning shows
- Tested changing when there are no annotations > warning does not show

**Tested Import & Export**
- Turned on positive grading
- Export assessment and check yml image
- Imported and checked that positive grading is one

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

**Comment by victorhuangwq** [comment]
- Add warning when switching between positive and negative if there's annotation
- Add an indication about whether it's positive or negative grading

**Comment by victorhuangwq** [comment]
- Add is_positive_grading to export assessments 
-

File: app/views/assessments/_edit_problems.html.erb [hunk 1]
```diff
@@ -26,3 +47,7 @@
 </table>
 <br>
 <%= link_to "Add Problem".html_safe, new_course_assessment_problem_path(@course, @assessment), { :class=> "btn" } %>
+
+<div class="action_buttons">
```
Comment (damianhxy) [inline app/views/assessments/_edit_problems.html.erb:51]: Saving brings us back to the basic tab. It seems that the `tab_index` method in `assessments_controller.rb` is missing the check for the problems tab.

i.e., we need to add
`elsif params[:problems]`
`tab_name = "problems"`

**Review by damianhxy — APPROVED** [review]
Functionally, works as expected

Left some comments, but otherwise lgtm!

File: app/assets/javascripts/edit_assessment.js [hunk 2]
```diff
@@ -25,5 +31,21 @@
...
       changes = false;
     });
+
+    $('input[name="assessment[is_positive_grading]"]').on('change', function() {
+      if(has_annotations){
+        if ($(this).prop('checked') !=  is_positive_grading) {
+          $('#grading-change-warning').show();
+        }
+        else{
+          console.log("not changed");
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply app/assets/javascripts/edit_assessment.js:41]: oh yes i will do that thank you

File: app/assets/stylesheets/style.css.scss [hunk 3]
```diff
@@ -972,6 +972,19 @@ form p:last-child {
   padding-right: 12px;
 }
 
+/**
+* Switch styling
+*/
+
+.switch > label > b{
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply app/assets/stylesheets/style.css.scss:979]: sounds good

File: app/form_builders/form_builder_with_date_time_input.rb [hunk 4]
```diff
@@ -55,6 +55,10 @@ def check_box(name, *args)
     # Materalize requires the label to be in a span
     field = super name, *(args + [options])
 
+    if(options[:default])
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply app/form_builders/form_builder_with_date_time_input.rb:58]: hmm! I'm wondering why didn't rubocop catch this. let me check

File: app/views/assessments/_edit_problems.html.erb [hunk 5]
```diff
@@ -1,4 +1,26 @@
-<br>
+<script type="text/javascript">
+	is_positive_grading = <%= @is_positive_grading %>;
+	has_annotations = <%= @has_annotations %>;
+</script>
+
+<ul class="collection" hidden id="grading-change-warning" >
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply app/views/assessments/_edit_problems.html.erb:6]: that make sense. I will change and see how it is

File: app/views/assessments/_edit_problems.html.erb (hunk 1 above)
Comment (victorhuangwq) [reply app/views/assessments/_edit_problems.html.erb:51]: let me look into that

File: app/views/assessments/_edit_problems.html.erb (hunk 5 above)
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply app/views/assessments/_edit_problems.html.erb:6]: image
I think it works

File: app/views/assessments/_edit_problems.html.erb (hunk 1 above)
Comment (victorhuangwq) [reply app/views/assessments/_edit_problems.html.erb:51]: fixed that

**Comment by damianhxy** [comment]
LGTM!

**Pull request merged**